adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -267,7 +267,7 @@ private void handleEndpointConnectionWithAuthentication(final MqttEndpoint endpo
         if (endpoint.auth() == null) {
             LOG.trace("no auth information in endpoint found");
             endpoint.reject(MqttConnectReturnCode.CONNECTION_REFUSED_BAD_USER_NAME_OR_PASSWORD);
-            LOG.info("connection request from client [{}] rejected [{}]",
+            LOG.debug("connection request from client [{}] rejected [{}]",
```

Comment:
This is one of the few cases where I would considering info level, to let massive bad credentials attack appear in the log.
Although checking security relevant attacks never can only be done by logging, of course - and intentionally filling up the logs by bad credentials could be used to make the "real attack" harder detectable.
So, just some thoughts to that problem - INFO or DEBUG both have good arguments...

In full production scenarios there probably

Reply:
if we want to be able to detect this, we should introduce a corresponding metric IMHO

Reply:
+1 - leave it at debug.
Metric for fraud detection sounds good to me, in production scenarios even more might be necessary (specialized intrusion detection software e.g.). Shouldn't be the scope here.